Sources/Swift/Sentry.swift

```diff
@@ -0,0 +1,3 @@
+//For some reason Xcode needs a swift file in a objc only target to be able to use swift classes from another target.
```

Comment:
If I just delete this file, I'm still able to build the Sentry target and run e.g. `SentryUIViewControllerPerformanceTrackerTests` (which uses `SwiftDescriptor.getObjectClassName`). What am I missing? I also set up a simple test project where I was able to call a swift function defined in a swift framework from an objc-only target.

Is this a SPM thing?

Reply:
I did a "Clean Project" but i will just delete the entire derived data folder, did you try either of those? Which scheme/destination are you building?

Reply:
Sentry Scheme.

I got this:
![image](https://user-images.githubusercontent.com/28265868/203499976-78c772f9-cfb5-46a7-a406-cef3721c41b5.png)

I tried it all. This was a huge problem for me because I couldn't make the Swift target work with the Obj-C target until I added this single Swift file inside the latter.

Reply:
I see it now after deleting derived data.

Reply:
If I add `${SDK_ROOT}/usr/lib/swift` to `LIBRARY_SEARCH_PATHS`, it resolves all but one, `Undefined symbol: __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency`. I just can't get it to find that one, even though I've found where it lives on disk:
```
❯ nm /Applications/Xcode14.1/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator/libswiftCompatibilityConcurrency.a

/Applications/Xcode14.1/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator/libswiftCompatibilityConcurrency.a(CompatibilityConcurrency.cpp.o):
...
0000000000000178 D __swift_FORCE_LOAD_$_swiftCompatibilityConcurrency
```
I wound up making a few other changes that almost get it there, to the point where it's failing to import the SentryPrivate module, something with the autogenerated -Swift.h header, but I can't see why it fails because it's just 
```
#if 0
... a whole bunch of stuff, including the SwiftDescriptor definition...
#else
#error unsupported Swift architecture
#endif
```
I wish I knew what was resolving to that `0` in the initial `#if` statement.

Maybe I'll add this info to https://forums.swift.org/t/could-not-find-or-use-auto-linked-library-swiftcompatibility50/54351/13

None of this should be necessary 🤔 Just mix Swift and ObjC, they said. It'll Just Work, they said 🤣